addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,182 @@
+odoo.define('website_event_track.website_event_track_proposal', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var publicWidget = require('web.public.widget');
+    var utils=require('web.utils');
+
+    var _t = core._t;
+
+    publicWidget.registry.websiteEventProposalForm = publicWidget.Widget.extend({
+
+    selector: '.o_website_event_track_proposal_form',
+    events: {
+        'input input[name="partner_name"]': '_onNameInputPropagateOnContact',
+        'click .o_wetrack_add_contact_information': '_onToggleContactInformationClick',
+        'click .o_wetrack_proposal_submit_button': '_send',
+    },
+
+    /**
+     * Check validity of input fields : empty required fields or bad formatted ones are 
+     * marked as is-invalid and o_has_error.
+     * 
+     * @private
+     */
+    _check_error_fields: function () {
```

Comment:
As check is a bit generic, would name it like _validate_form which is more accurate. It is just an example, I let you go through your code and see if naming can be improved :) .

Reply:
Thibault meant `_validateForm` of course, since we use camelCase in JS files :smile: